### PR TITLE
Feature/docs

### LINKED ISSUE #41: auto generate help info on settings like configurable rules fields, available messages

### LINKED ISSUE #64: Create external documentation on all features, user guide

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #41 
Closes #64

This documentation is far from being complete but it should be good starting point for us. Some data like available checkers and configurable parameters are loaded dynamically from source code. 
Documentation will be hosted on readthedocs (after merging this branch). 

I added bunch of docs, removed obsolete methods (previous version of disabling rule from source code) and renamed two checkers.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    app.connect("source-read", rstjinja)
+
+"""
+.. csv-table:: a title
+   :header: "name", "firstname", "age"
+   :widths: 20, 20, 10
+"""
+
+def get_checker_docs():
+    checker_docs = defaultdict(list)
```
Comment (bhirsz) [inline docs/conf.py:86]: this (kind of ugly) method loads information from existing checkers/rules. Thanks for this we don't need to manually change documentation where we add/update our rules.

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+# -- Project information -----------------------------------------------------
+
+project = 'Robocop'
+copyright = '2020, Bartlomiej Hirsz, Mateusz Nojek'
+author = 'Bartlomiej Hirsz, Mateusz Nojek'
+
+# The full version, including alpha/beta/rc tags
+# release = robocop.version
+relese = "0.0.1"
```
Comment (mnojek) [inline docs/conf.py:26]: typo: release

File: README.rst [hunk 3]
```diff
@@ -1,6 +1,13 @@
+.. image:: docs/images/robocop_logo.png
+   :scale: 50 %
```
Comment (mnojek) [inline README.rst:2]: Either scaling does not work or the logo is still too big. I would make it smaller :)

File: README.rst [hunk 4]
```diff
@@ -54,120 +61,18 @@ Robocop will execute all checks on provided list of paths::
...
+    * customized format of output message
+    * generating customized reports
+    * configurable lint rules
+    * output can be redirected to file
 
-  robocop --output robocop.log
+Documentation
+-------------
 
+Full documentation available .. _here: https://robocop.readthedocs.io/
```
Comment (mnojek) [inline README.rst:78]: This link does not display properly

File: README.rst [hunk 5]
```diff
@@ -54,120 +61,18 @@ Robocop will execute all checks on provided list of paths::
...
-    Info: info, i
-    Fatal: fatal, f
-
-Save output to file
--------------------
+Main features
+-------------
 
-You can redirect output of Robocop to a file by using pipes (``>`` in unix) or by ``-o`` / ``--output`` argument::
+    * including/excluding rules from command line or source code
```
Comment (mnojek) [inline README.rst:69]: I think the features should be written in the same way, e.g. 'output redirection to file'. Just an idea

File: README.rst (hunk 3 above)
Comment (bhirsz) [reply README.rst:2]: I've tested scaling with readthedocs, not readme (where I moved logo later). So scaling works but like you wrote I need to make it smaller

**Pull request merged**